navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/camera/NavigationCamera.java

```diff
@@ -0,0 +1,195 @@
+      mapboxMap.animateCamera(CameraUpdateFactory.newCameraPosition(cameraPosition), 2000,
+        new MapboxMap.CancelableCallback() {
+          @Override
+          public void onCancel() {
+
+          }
+
+          @Override
+          public void onFinish() {
+            navigation.addProgressChangeListener(NavigationCamera.this);
+          }
+        });
+    } else {
+      navigation.addProgressChangeListener(NavigationCamera.this);
+    }
+  }
+
+  public void resume(Location location) {
+    if (location != null) {
+      CameraPosition position = buildCameraPositionFromLocation(location);
+      mapboxMap.animateCamera(CameraUpdateFactory.newCameraPosition(position), 2000,
+        new MapboxMap.CancelableCallback() {
+          @Override
+          public void onCancel() {
+
+          }
+
+          @Override
+          public void onFinish() {
+            navigation.addProgressChangeListener(NavigationCamera.this);
+          }
+        });
+    } else {
+      navigation.addProgressChangeListener(NavigationCamera.this);
+    }
+  }
+
+  @Override
+  public void onProgressChange(Location location, RouteProgress routeProgress) {
+    this.location = location;
+    easeCameraToLocation(location);
+  }
+
+  public void setCameraTrackingLocation(boolean trackingEnabled) {
+    this.trackingEnabled = trackingEnabled;
+  }
+
+  public void resetCameraPosition() {
+    this.trackingEnabled = true;
+    if (location != null) {
+      LatLng target = new LatLng(location.getLatitude(), location.getLongitude());
+      CameraPosition cameraPosition = new CameraPosition.Builder()
+        .tilt(CAMERA_TILT)
+        .zoom(CAMERA_ZOOM)
+        .target(target)
+        .bearing(location.getBearing())
+        .build();
+      mapboxMap.easeCamera(CameraUpdateFactory.newCameraPosition(cameraPosition), 750, false);
+    }
+  }
+
+  private void easeCameraToLocation(Location location) {
+    CameraPosition cameraPosition = buildCameraPositionFromLocation(location);
+
+    if (trackingEnabled) {
+      mapboxMap.easeCamera(CameraUpdateFactory.newCameraPosition(cameraPosition), 1000, false);
+    }
+  }
+
+  @NonNull
+  private CameraPosition buildCameraPositionFromRoute(DirectionsRoute route) {
+    LineString lineString = LineString.fromPolyline(route.getGeometry(), Constants.PRECISION_6);
+
+    double initialBearing = TurfMeasurement.bearing(
+      Position.fromLngLat(
+        lineString.getCoordinates().get(0).getLongitude(), lineString.getCoordinates().get(0).getLatitude()
+      ),
+      Position.fromLngLat(
+        lineString.getCoordinates().get(1).getLongitude(), lineString.getCoordinates().get(1).getLatitude()
+      )
+    );
+
+    Position targetPosition = TurfMeasurement.destination(
+      Position.fromCoordinates(
+        lineString.getCoordinates().get(0).getLongitude(), lineString.getCoordinates().get(0).getLatitude()
+      ),
+      120, initialBearing, TurfConstants.UNIT_METERS
```

Comment:
this 120 should become a constant

Reply:
`ScreenDensityMap` in https://github.com/mapbox/mapbox-navigation-android/pull/198/commits/83dd2cf3b22eb8f99a1a16f258526c5f7f3c29fd will address this